docs/source/apidocs/wavefunction.rst

```diff
@@ -16,15 +19,13 @@ Wavefunction Methods
 
 
 .. currentmodule:: pyquil.wavefunction
-
 .. autoclass:: pyquil.wavefunction.Wavefunction
 
     .. rubric:: Attributes and Methods
     .. autosummary::
         :toctree: autogen
         :template: autosumm.rst
 
-        ~Wavefunction.amplitudes
```

Comment:
`Wavefunction.amplitudes` appears to have gone away. Note that the list of attributes and methods here (and above for `WavefunctionSimulator`) is far from complete. Not sure if intentional or just out-of-sync with current class definitions.

Reply:
Is there a way to auto-regenerate these instead of deleting the bad items by hand? (Or did you do that already?)

Reply:
An excellent question! Basically everything I know about Sphinx, I learned in the last 24 hours, so I might be missing something, but my understanding is that we use the [autodoc extension](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#module-sphinx.ext.autodoc) plus a config file setting [`autosummary_generate`](https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html#confval-autosummary_generate) to do automatic doc generation, which I guess is more-or-less equivalent to using the command-line tool [`sphinx-autogen`](https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html#sphinx-autogen-generate-autodoc-stub-pages).

I think these auto-generated docs stubs typically just expand to a simple header + `.. autoclass::` though, and the result is a somewhat bare-bones description that only includes the class, class-level docstring, and `__init__` parameters + docstring. Whereas I assume the point of these "hand rolled" rst files is provide a bit more context and include some of the more useful / common methods. But I am guessing here.

You can actually see an example of the autodoc-generated stub vs. a hand-crafted class summary for the `WavefunctionSimulator` just above this (viz. my comments above about the `:noindex:` change). The first link is to the autodoc-generated stub, the second is our lovingly hand-crafted version from above, which includes some methods as well.

http://docs.rigetti.com/en/stable/apidocs/autogen/pyquil.api.WavefunctionSimulator.html#pyquil.api.WavefunctionSimulator

http://docs.rigetti.com/en/stable/apidocs/wavefunction.html#pyquil.api.WavefunctionSimulator

Now I think if we wanted to, we could update the `autosumm.rst` template that is being used here to automatically include all the "public" class [methods](http://www.sphinx-doc.org/es/stable/ext/autosummary.html#methods) or attributes, but that would then apply to all the `autosummary`-generated tables in this `apidocs` subdirectory (or anywhere else we use the `autosumm.rst` template). My assumption was that these method lists are intentionally hand-curated, but I could be wrong about that.

Reply:
I just noticed that `Wavefunction.amplitudes` has not actually disappeared, but is merely an attribute of the class.

Reply:
Hmmm... it looks like maybe [`autoclass`' `:member:` directive](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-autoclass) could be used to auto-generate the list of "public" members without the need to fiddle with `autosumm.rst`, assuming we want to go down that route.

Note this is conjecture based on a sphinx doc search; haven't actually tested it yet.